### DISCUSSION TITLE
Add Test262 for `ecma_transforms_testing`

### DISCUSSION THREAD

**Discussion by magic-akari** [swc-project/swc#3533] [General]
I am trying to add test262 test cases to swc.
But I ran into some issues that made me consider whether it was worth it.

What am I trying to do:

- add [test262](https://github.com/tc39/test262) as swc git submodule
- add [test262-harness](https://github.com/bterlson/test262-harness) as test runner
- set some transform targets like es2020, es2017, safari 11, chrome 46, etc

The issues I encountered：

1. Test cases are broken without any transformation.
2. It cost me 25 minutes to run all the `test/language` cases without any transformation.

I pulled the test262 repo and ran the test cases directly.
I tested them on node 16 and 5% of the test cases failed. This is a predictable result since V8 did not implement all the features according to the spec.
If this test262 is being considered for use , then you need to know if it was originally passed or not.
Some test cases will fail in the parsing phase. This means that we may get the Error directly instead of the transformed files. It is a issue about how to match the errors as well.

I ran the test case on my MacBook (13-inch, M1, 2020).
There were no transformation, only one round, but it took 25 minutes to finish the test cases, which is too long for me.
The translation time needs to be counted for as well.

What do you think about it?
Is it worth introducing test262?

**Comment by kdy1** [comment]
I agree that it's too long. 
If it takes 25 minutes on local, it will take more than an hour on CI.
We have various targets, so we may have to run those multiple times, with different targets.


I think it does not worth adding.
